RevenueCatUI/PaywallView.swift

```diff
@@ -239,6 +250,44 @@ struct LoadedOfferingPaywallView: View {
         }
     }
 
+    private var eventData: PaywallEvent.Data {
+        self.updateSessionIfNeeded()
+
+        return .init(
+            offering: self.offering,
+            paywall: self.paywall,
+            sessionID: self.session.id,
+            displayMode: self.mode,
+            locale: .current,
+            darkMode: self.colorScheme == .dark
+        )
+    }
+
+    private func updateSessionIfNeeded() {
+        let newPaywall: DisplayedPaywall = .init(offering: self.offering, paywall: self.paywall)
+        guard self.session.lastPaywall != newPaywall else { return }
+
+        self.session.lastPaywall = newPaywall
+        self.session.id = .init()
```

Comment:
Note: we're baking in an assumption here: 
We're saying that if SwiftUI keeps the PaywallView in memory but calls onAppear more than once, we consider it to be the same session, right? 
If that's the case, maybe we should explicitly call it out here?

Reply:
Yup, I'll document it (I've also confirmed with @joshdholtz that repeated view events will be handled in the backend).